File: src/dataset/cpp/grains_exercism-Akshive/gemini/GeminiFlash_grains_v3.cpp
#include "grains.h"

namespace grains {

unsigned long long square(int sq) {
  if (sq < 1 || sq > 64) {
    return 0;
  }
  return (1ULL << (sq - 1));
}

unsigned long long total() {
  return ~0ULL;
}

} // namespace grains

File: src/dataset/typescript/grains_exercism-wackerow/gemini/GeminiFlash_grains_v3.ts
export default class Grains {
  public static square(digit: number): bigint {
    if (digit <= 0 || digit > 64 || digit % 1 !== 0) {
      throw 'Chessboard location must be integer from 1 to 64.';
    }
    return 1n << (BigInt(digit) - 1n);
  }

  public static total(): bigint {
    return (1n << 64n) - 1n;
  }
}